File: Java/Medium/Accepted/queries-on-a-permutation-with-key_4.java
class Solution {
    public int[] processQueries(int[] queries, int m) {
        
        int[] result = new int[queries.length];
        int[] P = new int[m];
        for (int i = 0; i < m; i++) {
            P[i] = i + 1;
        }
        for (int i = 0; i < queries.length; i++) {
            int pos = 0;
            for (int j = 0; j < m; j++) {
                if (P[j] == queries[i]) {
                    pos = j;
                    break;
                }
            }
            result[i] = pos;
            int temp = P[pos];
            for (int j = pos; j > 0; j--) {
                P[j] = P[j - 1];
            }
            P[0] = temp;
        }
        return result;
    }
}

File: Rust/Medium/Accepted/queries-on-a-permutation-with-key_4.rs
impl Solution {
    pub fn process_queries(queries: Vec<i32>, m: i32) -> Vec<i32> {
        
        let mut p = (1..=m).collect::<Vec<i32>>();
        let mut res = Vec::new();
        for i in queries {
            let index = p.iter().position(|&x| x == i).unwrap();
            res.push(index as i32);
            p.remove(index);
            p.insert(0, i);
        }
        res
    }
}